pkgs/activitywatch/default.nix

```diff
@@ -185,33 +179,36 @@ rec {
 
     nativeBuildInputs = [
       python3.pkgs.poetry
-      python3.pkgs.pyqt5 # for pyrcc5
-      libsForQt5.wrapQtAppsHook
-      xdg-utils
+      qt6.wrapQtAppsHook
     ];
 
     propagatedBuildInputs = with python3.pkgs; [
       aw-core
-      pyqt5
+      qt6.qtbase # Needed for qt6.wrapQtAppsHook
+      qt6.qtsvg # Rendering icons in the trayicon menu
+      pyqt6
       click
     ];
 
     # Prevent double wrapping
     dontWrapQtApps = true;
 
     postPatch = ''
-      sed -E 's#click = "\^7.1.2"#click = "^8.0"#g' -i pyproject.toml
-      sed -E 's#PyQt5 = "5.15.2"#PyQt5 = "^5.15.2"#g' -i pyproject.toml
-    '';
-
-    preBuild = ''
-      HOME=$TMPDIR make aw_qt/resources.py
+      sed -E 's#PyQt6 = "6.3.1"#PyQt6 = "^6.4.0"#g' -i pyproject.toml
     '';
 
     postInstall = ''
-      install -Dt $out/etc/xdg/autostart resources/aw-qt.desktop
-      xdg-icon-resource install --novendor --size 32 media/logo/logo.png activitywatch
-      xdg-icon-resource install --novendor --size 512 media/logo/logo.png activitywatch
+      install -D resources/aw-qt.desktop $out/share/applications/aw-qt.desktop
+      install -D resources/aw-qt.desktop $out/etc/xdg/autostart/aw-qt.desktop
```

Comment:
We may want to add a toggle for the autostart here. I'm also having an issue on KDE where the application launches before KDE has finished loading its panel with the taskbar. Perhaps we should have the user setup autostart themselves, allow them to toggle it, or in the future when we have it in `nixpkgs` add an activitywatch NixOS module to enable a systemd service.

Reply:
Weird, it worked just fine in GNOME Shell. I would expect autostart to work reliably across different environment. I do not think that there is much of a point of decoupling installation from autostart, though systemd service might be nicer, as it is more customizable.